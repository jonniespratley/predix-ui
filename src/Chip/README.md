The `<Chip/>` component is a persistent object that can be toggled on and off.
The static chip is not meant to be dismissed or removed. A Static Chip is best used for curated filters.


## Usage

```react
const {Chip} = PxReact;
<Chip showIcon selected>
  Text Label
</Chip>
```


## Properties

```table
span: 6
rows:
  - Name: children
    Type: node
    Description: This contents of the chip.
  - Name: showIcon
    Type: bool
    Description: Toggle the icon on or off.
```


### Styling
The following custom properties are available for styling:

```code
rows:
  - Custom Property: --px-chip-background-color
    Description: Background color for chip

  - Custom Property: --px-chip-background-color--hover
    Description: Background color for chip on hover

  - Custom Property: --px-chip-background-color--selected
    Description: Background color for chip on selected
    
  - Custom Property: --px-chip-border-color
    Description: Border or outline color for chip

  - Custom Property: --px-chip-border-color--hover
    Description: Border or outline color for chip on hover

  - Custom Property: --px-chip-border-color--selected
    Description: Border or outline color for chip on selected

  - Custom Property: --px-chip-icon-color
    Description: Icon color

  - Custom Property: --px-chip-icon-color--hover
    Description: Icon color on hover

  - Custom Property: --px-chip-text-color
    Description: Text color for chip content

  - Custom Property: --px-chip-text-color--hover
    Description: Text color for chip content on hover

  - Custom Property: --px-chip-text-color--selected
    Description: Text color for chip content on selected

  - Custom Property: --px-chip-max-width
    Description: Max width of the chip (default: 120px)
```
