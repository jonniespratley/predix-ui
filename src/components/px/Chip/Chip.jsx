import React from 'react';
import PropTypes from 'prop-types';
import styled, { css } from 'styled-components';

const Chip = styled.div`
  display         : inline-block;
  align-items     : center;
  height          : 20px;
  padding-left    : 10px;
  padding-right   : 10px;
  background-color: var(--px-chip-background-color, white);
  border          : 1px solid var(--px-chip-border-color, gray);

  &:hover {
    cursor          : pointer;
    background-color: var(--px-chip-background-color--hover, rgba(gray, 0.2));
    border          : 1px solid var(--px-chip-border-color--hover, gray);
  }

  ${props => props.selected && css`
    background-color: var(--px-chip-background-color--selected, white);
    border          : 1px solid var(--px-chip-border-color--selected, blue);
  `}

`;

const ChipContent = styled.div`
  line-height: 18px;
  display    : inline-block;
  min-width  : 0;
  max-width  : var(--px-chip-max-width, 120px);
  color      : var(--px-chip-text-color, gray);
`;

const ChipIcon = styled.div`
  height: 12px;
  width : 12px;
  &:hover{
    color: var(--px-chip-icon-color--hover, blue);
  }
`;

/**
 * Chip component
 */
const Component = ({
  icon,
  selected,
  showIcon,
  children
}) => (
  <Chip selected={selected}>
    <ChipContent>
      {children}
      {showIcon && <ChipIcon>{icon}</ChipIcon>}
    </ChipContent>
  </Chip>

);

Component.defaultProps = {
  icon: null,
  selected: null,
  showIcon: false,
  children: null
};

Component.propTypes = {
  icon: PropTypes.string,
  selected: PropTypes.bool,
  showIcon: PropTypes.bool,
  children: PropTypes.node
};

export default Component;
