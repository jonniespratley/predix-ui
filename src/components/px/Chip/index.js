import React from 'react';
import classnames from 'classnames';

import stylesheet from './px-chip.scss';

/**
 * Chip component
 */
export default ({
  style,
  label = 'Chip',
  icon,
  selected,
  showIcon,
  children
}) => {

  const baseClasses = classnames('px-chip');
  const chipClasses = classnames('chip', 'zeta', {'selected': selected});

  return (
    <div className={baseClasses} style={style}>
      <div className={chipClasses}>
        <div className='flexContainer'>
          <span className='chip__content truncate'>{children}</span>
          {showIcon && <span className='chip__actionable-icon'>{icon}</span>}
        </div>
      </div>
      <style jsx>{stylesheet}</style>
    </div>
  );
}
