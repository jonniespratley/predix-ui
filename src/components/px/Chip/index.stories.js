import React from 'react';
import { storiesOf } from '@storybook/react';
import { withKnobs, text, boolean } from '@storybook/addon-knobs';

import Chip from './';

storiesOf('Chip', module)
  .addDecorator(withKnobs)
  .add('default', () => (
    <Chip showIcon={boolean('showIcon', false)}>{text('children', 'Text Label')}</Chip>
  ));
