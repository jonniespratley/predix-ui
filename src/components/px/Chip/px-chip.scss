// Generic
//@import "px-normalize-design/_generic.normalize.scss";
//@import "px-box-sizing-design/_generic.box-sizing.scss";
@import "px-helpers-design/_generic.helpers.scss";


/**
 * Chip styles
 */
.px-chip {

  // Component

  .chip {
    display         : inline-block;
    align-items     : center;
    height          : calculateRem(20px);
    padding-left    : calculateRem(10px);
    padding-right   : calculateRem(10px);
    background-color: var(--px-chip-background-color, white);
    border          : 1px solid var(--px-chip-border-color, gray);

    &:hover {
      cursor          : pointer;
      background-color: var(--px-chip-background-color--hover, rgba(gray, 0.2));
      border          : 1px solid var(--px-chip-border-color--hover, gray);

      .chip__actionable-icon {
        color: var(--px-chip-icon-color--hover, blue);
      }
    }
  }

  .chip__actionable-icon {
    height: calculateRem(12px);
    width : calculateRem(12px);
  }

  :host(:focus) {
    outline: none;
  }

  .selected {
    background-color: var(--px-chip-background-color--selected, white);
    border          : 1px solid var(--px-chip-border-color--selected, blue);
  }

  .chip[show-icon] {
    padding-right: 5px;
  }

  .flexContainer {
    display        : flex;
    justify-content: center;
    align-items    : center;
  }

  .chip__content {
    line-height: calculateRem(18px);
    display    : inline-block;
    min-width  : 0;
    max-width  : var(--px-chip-max-width, calculateRem(120px));
    color      : var(--px-chip-text-color, gray);
  }

  .chip__content[selected] {
    color: var(--px-chip-text-color--selected, blue);
  }

  .chip__actionable-icon {
    color      : var(--px-chip-icon-color, gray);
    margin-left: calculateRem(5px);
  }

}
