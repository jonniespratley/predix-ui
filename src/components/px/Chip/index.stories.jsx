import React from 'react';
import { storiesOf } from '@storybook/react';
import { withKnobs, text, boolean } from '@storybook/addon-knobs';
import { withReadme } from 'storybook-readme';

import README from './README.md';
import Chip from './';

storiesOf('Components / Chip', module)
  .addDecorator(withKnobs)
  .addDecorator(withReadme(README))
  .add('default', () => (
    <Chip showIcon={boolean('showIcon', false)}>{text('children', 'Text Label')}</Chip>
  ));
