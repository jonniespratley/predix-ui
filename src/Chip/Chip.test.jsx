import { expect } from 'chai';
import React from 'react';
import { shallow } from 'enzyme';

import Chip from './Chip';

describe('Chip', () => {
  test('should render with text', () => {
    const wrapper = shallow(<Chip>Label</Chip>);
    expect(wrapper).matchSnapshot();
  });
  test('should render with selected', () => {
    const wrapper = shallow(<Chip selected>Label</Chip>);
    expect(wrapper).matchSnapshot();
  });
  test('should render with icon', () => {
    const wrapper = shallow(
      <Chip showIcon icon="px-utl:close">Label</Chip>
    );
    expect(wrapper).matchSnapshot();
  });
});
