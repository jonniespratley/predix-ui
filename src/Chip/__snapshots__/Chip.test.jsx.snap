// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Chip should render with icon 1`] = `
<styled.div
  selected={null}
>
  <styled.div>
    Label
    <styled.div>
      <Icon
        className={null}
        color="inherit"
        icon="px-utl:close"
        size={16}
        style={null}
        viewBox={null}
      />
    </styled.div>
  </styled.div>
</styled.div>
`;

exports[`Chip should render with selected 1`] = `
<styled.div
  selected={true}
>
  <styled.div>
    Label
  </styled.div>
</styled.div>
`;

exports[`Chip should render with text 1`] = `
<styled.div
  selected={null}
>
  <styled.div>
    Label
  </styled.div>
</styled.div>
`;
